src/borg/archiver.py

```diff
@@ -1111,35 +1111,45 @@ def contents_changed(item1, item2):
         def compare_content(path, item1, item2):
             if contents_changed(item1, item2):
                 if item1.get('deleted'):
-                    return 'added {:>13}'.format(format_file_size(sum_chunk_size(item2)))
+                    sz = sum_chunk_size(item2)
+                    return ({"type": "added", "size": sz},
+                        'added {:>13}'.format(format_file_size(sz)))
                 if item2.get('deleted'):
-                    return 'removed {:>11}'.format(format_file_size(sum_chunk_size(item1)))
+                    sz = sum_chunk_size(item1)
+                    return ({"type": "removed", "size": sz},
+                        'removed {:>11}'.format(format_file_size(sz)))
                 if not can_compare_chunk_ids:
-                    return 'modified'
+                    return ({"type": "modified"}, "modified")
                 chunk_ids1 = {c.id for c in item1.chunks}
                 chunk_ids2 = {c.id for c in item2.chunks}
                 added_ids = chunk_ids2 - chunk_ids1
                 removed_ids = chunk_ids1 - chunk_ids2
                 added = sum_chunk_size(item2, added_ids)
                 removed = sum_chunk_size(item1, removed_ids)
-                return '{:>9} {:>9}'.format(format_file_size(added, precision=1, sign=True),
-                                            format_file_size(-removed, precision=1, sign=True))
+                return ({"type": "modified", "added": added, "removed": removed},
+                    '{:>9} {:>9}'.format(format_file_size(added, precision=1, sign=True),
+                    format_file_size(-removed, precision=1, sign=True)))
 
         def compare_directory(item1, item2):
             if item2.get('deleted') and not item1.get('deleted'):
-                return 'removed directory'
+                return ({"type": 'removed directory'}, 'removed directory')
             elif item1.get('deleted') and not item2.get('deleted'):
-                return 'added directory'
+                return ({"type": 'added directory'}, 'added directory')
 
         def compare_owner(item1, item2):
             user1, group1 = get_owner(item1)
             user2, group2 = get_owner(item2)
             if user1 != user2 or group1 != group2:
-                return '[{}:{} -> {}:{}]'.format(user1, group1, user2, group2)
+                return ({"type": "owner", "olduser": user1, "oldgroup": group1,
+                        "newuser": user2, "newgroup": group2},
```

Comment:
is this due to your pep8 checker being on 80?

would look better in one line and the project one's is at 120 chars per line iirc.

maybe check / revise that for your whole change.

Reply:
sorry, I have a habit of going to 80 to make side by side panels easier.

Reply:
ok, that's a use case at least.